### ISSUE TITLE
The intro freezes and so does the ingame (Linux Endeavour OS)

### ISSUE THREAD

**Issue by orderorder** [HarbourMasters/Starship#45]
Linux Endeavour OS

gcc versión 14.2.1 20240910 (GCC) 

Starship log:

20:04:37.636] [CrashHandler.cpp:72] [critical] Signal: 8
ERRONEUS ARITHEMETIC OPERATION
Registers:
RAX: 0x000000008000000A
RDI: 0x0000000080007224
RSI: 0x0000000000000000
RDX: 0x00000000FFFFFFFF
RCX: 0x000064ED021D10A0
R8:  0x0000000000000008
R9:  0x000064ED01FE7FA4
R10: 0x0000000000000000
R11: 0x0000000000000246
RSP: 0x00007FFD2255ABF8
RBX: 0x000064ED021D10A0
RBP: 0x000064ED021D0D40
R12: 0x0000000000000001
R13: 0x0000000000000000
R14: 0x00007D51E756E000
R15: 0x000064ED01CFF378
RIP: 0x000064ED01A36E5A
EFL: 0x0000000000010286
Traceback:
1 /usr/lib/libc.so.6(+0x3d1d0) [0x7d51e6c4c1d0]
2 osRecvMesg (+0x2A)
3 Controller_ReadData (+0x8E)
4 Graphics_ThreadUpdate (+0x145)
5 main (+0x35)
6 /usr/lib/libc.so.6(+0x25e08) [0x7d51e6c34e08]
7 __libc_start_main (+0x8C)
8 _start (+0x25)

**Comment by JimBimBum** [comment]
`/home/<user>/starship/build-cmake/torch/src/torch-build/torch otr [OPTIONS] <baserom.z64>`

you forgot to do this after, it'll fix it.

**Comment by metchebe** [comment]
Same issue as OP on Ubuntu Jammy (22.04), built from source.

In the intro, it crashes when the arwings are about to take off, when the engines start glowing.

**Comment by orderorder** [comment]
> `/home/<user>/starship/build-cmake/torch/src/torch-build/torch otr [OPTIONS] <baserom.z64>`
> 
> you forgot to do this after, it'll fix it.

it hasn't worked for me

**Comment by Melechtna** [comment]
> > `/home/<user>/starship/build-cmake/torch/src/torch-build/torch otr [OPTIONS] <baserom.z64>`
> > you forgot to do this after, it'll fix it.
> 
> it hasn't worked for me
> 
> [03:45:18.838] [CrashHandler.cpp:72] [critical] Signal: 11 INVALID ACCESS TO STORAGE Registers: RAX: 0x00024D6C40E5A065 RDI: 0x000055773F852DC0 RSI: 0x0000000000000010 RDX: 0x0000000000000000 RCX: 0x0000000000000001 R8: 0x0000000000000001 R9: 0x000055776AB2CEF0 R10: 0x000055776B0868C8 R11: 0x000055776AB2D044 RSP: 0x00007FFCB13FE4F8 RBX: 0x000055773F3E8780 RBP: 0x000055773F852E40 R12: 0x0000000000000001 R13: 0x0000000000000000 R14: 0x00007FF9DB532000 R15: 0x000055773F37F678 RIP: 0x000055773F071E4F EFL: 0x0000000000010212 Traceback: 1 /usr/lib/libc.so.6(+0x3d1d0) [0x7ff9dac4c1d0] 2 osSendMesg (+0x1F) 3 Graphics_ThreadUpdate (+0x145) 4 main (+0x35) 5 /usr/lib/libc.so.6(+0x25e08) [0x7ff9dac34e08] 6 __libc_start_main (+0x8C) 7 _start (+0x25)

I have nearly the exact same crash
```
[02:01:39.629] [ResourceLoader.cpp:81] [error] Failed to load resource: File not loaded
[02:01:39.629] [ResourceManager.cpp:163] [error] Failed to load resource file at path __OTR__ast_common/aBallDL
[02:01:44.741] [CrashHandler.cpp:72] [critical] Signal: 11
INVALID ACCESS TO STORAGE
Registers:
RAX: 0x0000000000000020
RDI: 0x0000000000000020
RSI: 0x0000000000000000
RDX: 0x0000000000000000
RCX: 0x00007FFCF00C6F90
R8:  0x00005895270CA010
R9:  0x0000000000000007
R10: 0x000058952923ED90
R11: 0x6694629F1A0E563B
RSP: 0x00007FFCF00C6F50
RBX: 0x0000589527115880
RBP: 0x00007FFCF00C6F50
R12: 0x0000000000000001
R13: 0x0000000000000000
R14: 0x000072AE274E1000
R15: 0x000058950EFD26D8
RIP: 0x000058950E74BAC4
EFL: 0x0000000000010202
Traceback:
1 /usr/lib/libc.so.6(+0x3d1d0) [0x72ae26c1e1d0]
2 std::vector<Gfx, std::allocator<Gfx> >::operator[](unsigned long) (+0x10)
3 gSPDisplayList (+0x151)
4 Display_ArwingWingTrail_Draw (+0x2F1)
5 Display_ArwingWingTrail_Update (+0x214)
6 Display_Update (+0xDCE)
7 Game_Draw (+0x91)
8 Game_Update (+0xB59)
9 Graphics_ThreadUpdate (+0x12D)
10 push_frame() (+0x9)
11 main (+0x39)
12 /usr/lib/libc.so.6(+0x25e08) [0x72ae26c06e08]
13 __libc_start_main (+0x8C)
14 _start (+0x25)
```

**Comment by metchebe** [comment]
> In the intro, it crashes when the arwings are about to take off, when the engines start glowing.

As suggested in https://github.com/HarbourMasters/Starship/issues/64#issuecomment-2561954339, by setting FPS to 30 I can get through the intro without the 'ERRONEUS ARITHEMETIC OPERATION' crash.

**Comment by orderorder** [comment]
> > In the intro, it crashes when the arwings are about to take off, when the engines start glowing.
> 
> As suggested in [#64 (comment)](https://github.com/HarbourMasters/Starship/issues/64#issuecomment-2561954339), by setting FPS to 30 I can get through the intro without the 'ERRONEUS ARITHEMETIC OPERATION' crash.

Yes, I can confirm that, if we set it to 30 frames it doesn't crash.
